File: 94-binary-tree-inorder-traversal/94-binary-tree-inorder-traversal.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func inorderTraversal(root *TreeNode) []int {
    left, right := []int{}, []int{}
    if root == nil{
        return []int{}
    }else if root.Left != nil{
        left = inorderTraversal(root.Left)
    }
    
    left = append(left, root.Val)
    
    if root.Right != nil{
        right = inorderTraversal(root.Right)
    }
    
    return append(left, right...)
}

File: 94-binary-tree-inorder-traversal/94-binary-tree-inorder-traversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    res = None
    def dfs(self, root):
        if not root:
            return
        self.dfs(root.left)
        self.res.append(root.val)
        self.dfs(root.right)
        
    def inorderTraversal(self, root: Optional[TreeNode]) -> List[int]:
        self.res = []
        self.dfs(root)
        return self.res
                